Replace `appversion.{app_id}` by `current_version.compatible_apps.{app_id}` in ES

diox opened the pull request:
Fix #3927

EnTeQuAk commented:
Are the changed lines already covered by tests, e.g break if you just remove things?

r+ otherwise.

diox commented:
Yeah, the code in `search/views.py` is properly covered. I removed the `legacy_api/utils.py` function I modified entirely since it was not used anywhere.
